src/sethostid/sethostid.c

```diff
@@ -0,0 +1,92 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+/*
+ * Copyright 2019 Joyent, Inc.
+ */
+
+/*
+ * This program sets the hostid based on a crc32 of the system's UUID string.
+ * Normally, a Linux machine sets the hostid based on the system's IP address or
+ * based on a random number if the IP addess is not known.
+ *
+ * This is useful on live media that would like to import a zfs pool so that
+ * pool-in-use detection doesn't get tripped up across reboots.
+ */
+
+#include <errno.h>
+#include <fcntl.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <string.h>
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <unistd.h>
+#include <zlib.h>
+
```

Comment:
Dunno, this was written by Mike.

I ported it to rust as a play project one Saturday just for the practice. I left the C one in the repo because I'm not ready to outright delete it. The rust one has been in all the PI builds for the past several weeks, so this is mostly vestigial at this point.

For me, the rust one is more maintainable because I know rust better, but other than that there's not really a strong reason to prefer rust over C for this. The main thing is that for CNs that already have a pool, if the output of this command changes it'll make the pool not importable (zpool import will consider it foreign until it's manually imported).

Reply:
To be clear, sethostid and sethostid.rs are basically the same, just one is C and the other is Rust. The output should be identical in all cases.

Reply:
Gotcha!